### DISCUSSION TITLE
rendering localized shapes from id with liquid - what is wrong?

### DISCUSSION THREAD

**Discussion by bashuss** [OrchardCMS/OrchardCore#11908] [Q&A] [answered]
My current Orchard Core Version is 1.3

I am trying to render menus by id into a dedicated zone of my layout.liquid.
All is derived from the agency theme, but the LandingPage is localized LocalizationPart and supports adding of a selected menu item (content picker field). The Menu Items are also localized with LocalizationPart and the correct localization is picked by the content picker (which does not prefilter shown items to the current localization of the landing page).

In the Preview of my localized LandingPage the correct localized menu never renders. 
I can retrieve the correct localized Menu from `menuid` (that can be seen in the HTML comment I render), which I'd like to render into my zone by the following code. 
```
{% assign menu = Content.ContentItemId[menuid] %}
{% shape_type menu "Menu" %}
{% shape_display_type menu "Summary" %}
{% shape_cache menu cache_id: menu.ContentItem.Content.AliasPart.Alias, cache_expires_after: "00:05:00" %}
<!-- MENUE alternative: {{ menu.ContentItem.Content | json | raw }} -->
{% zone "Menu" %}
{{ menu | shape_render }}
{% endzone  %}
```
The used code for rendering the new menu shape is this:
```
<div class="collapse navbar-collapse" id="navbarResponsive">
    <!-- template model: {{ Model.ContentItem.Content | json | raw }} -->
    <ul class="navbar-nav text-uppercase ml-auto">
        {% for item in Model.Items %}
        {{ item | shape_render }}
        {% endfor %}
    </ul>
</div>
```
Also within the Menu-Template, I do render an HTML comment, but here I get the Data for standard language of the localized menu instead of the one, which is rendered in the other HTML comment. So the content of the shape changes during rendering from one language to another!? What is wrong?

Here both HTML comments:

outside comment:
`<!-- MENUE alternative: {"MenuPart":{},"TitlePart":{"Title":"Main Menu"},"MenuItemsListPart":{"MenuItems":[{"ContentItemId":"4mkabcxvfbeqm4vp45k0amn55p","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Home","Latest":false,"Published":false,"ModifiedUtc":null,"PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":null,"LinkMenuItemPart":{"Name":"Home","Url":"~/"}},{"ContentItemId":"4vbfe30jrshafscxtv7hrwyzw1","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Offers","Latest":false,"Published":false,"ModifiedUtc":"2021-04-07T11:31:19.4861638Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","LinkMenuItemPart":{"Name":"Offers","Url":"~/#services"},"LinkMenuItem":{}},{"ContentItemId":"44tb0p3z1kqsh5ka458gppb7n8","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Sessions","Latest":false,"Published":false,"ModifiedUtc":"2021-04-09T20:07:52.3548396Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","LinkMenuItemPart":{"Name":"Sessions","Url":"~/#portfolio"},"LinkMenuItem":{}},{"ContentItemId":"4nj7tz63c5mh3spzqb54btnjn1","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"About","Latest":false,"Published":false,"ModifiedUtc":null,"PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":null,"LinkMenuItemPart":{"Name":"About","Url":"~/#about"},"LinkMenuItem":{}},{"ContentItemId":"40mmetjhkn04b6ev9cxjjgdtcj","ContentItemVersionId":null,"ContentType":"ContentMenuItem","DisplayText":"Blog","Latest":false,"Published":false,"ModifiedUtc":"2022-04-22T18:56:19.818449Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","ContentMenuItemPart":{"Name":"Blog","SelectedContentItem":{"ContentItemIds":["4t0ndmcqb10x81y8ee6f7h9m4y"]}},"ContentMenuItem":{}},{"ContentItemId":"4fhbsbcx6h0e72aj0r33y3haxq","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Contact","Latest":false,"Published":false,"ModifiedUtc":"2021-04-07T11:32:22.0515189Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","LinkMenuItemPart":{"Name":"Contact","Url":"~/#contact"},"LinkMenuItem":{}}]},"AliasPart":{"Alias":"main-menu-en"},"LocalizationPart":{"LocalizationSet":"40dhnchb7n9f32cbjfg47qabxt","Culture":"en"}} -->`

inside comment:
`<!-- template model: {"MenuPart":{},"TitlePart":{"Title":"Hauptmenü"},"MenuItemsListPart":{"MenuItems":[{"ContentItemId":"4mkabcxvfbeqm4vp45k0amn55p","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Home","Latest":false,"Published":false,"ModifiedUtc":null,"PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":null,"LinkMenuItemPart":{"Name":"Home","Url":"~/"}},{"ContentItemId":"4vbfe30jrshafscxtv7hrwyzw1","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Angebote","Latest":false,"Published":false,"ModifiedUtc":"2021-04-07T11:31:19.4861638Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","LinkMenuItemPart":{"Name":"Angebote","Url":"~/#services"},"LinkMenuItem":{}},{"ContentItemId":"44tb0p3z1kqsh5ka458gppb7n8","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Sitzungen","Latest":false,"Published":false,"ModifiedUtc":"2021-04-09T20:07:52.3548396Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","LinkMenuItemPart":{"Name":"Sitzungen","Url":"~/#portfolio"},"LinkMenuItem":{}},{"ContentItemId":"4nj7tz63c5mh3spzqb54btnjn1","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Über","Latest":false,"Published":false,"ModifiedUtc":null,"PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":null,"LinkMenuItemPart":{"Name":"Über","Url":"~/#about"},"LinkMenuItem":{}},{"ContentItemId":"40mmetjhkn04b6ev9cxjjgdtcj","ContentItemVersionId":null,"ContentType":"ContentMenuItem","DisplayText":"Blog","Latest":false,"Published":false,"ModifiedUtc":"2022-04-22T18:56:19.818449Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","ContentMenuItemPart":{"Name":"Blog","SelectedContentItem":{"ContentItemIds":["4t0ndmcqb10x81y8ee6f7h9m4y"]}},"ContentMenuItem":{}},{"ContentItemId":"4fhbsbcx6h0e72aj0r33y3haxq","ContentItemVersionId":null,"ContentType":"LinkMenuItem","DisplayText":"Kontakt","Latest":false,"Published":false,"ModifiedUtc":"2021-04-07T11:32:22.0515189Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"hussi","LinkMenuItemPart":{"Name":"Kontakt","Url":"~/#contact"},"LinkMenuItem":{}}]},"AliasPart":{"Alias":"main-menu"},"LocalizationPart":{"LocalizationSet":"40dhnchb7n9f32cbjfg47qabxt","Culture":"de-DE"}} -->`

Both menu translations have seperate aliases: `main-menu` and `main-menu-en` which are used for caching.

**Comment by Skrypt** [comment, marked as answer]
You need to simply create 2 different menus without using a LocalizationPart. It's unnecessary to localize the MenuItems if you can have 2 different menus that can be retrieved with an Alias with a culture suffix / prefix.

**Reply by bashuss** [reply]
ok, I'll try that and report - sounds like it should do.
I just thought, that it might be the correct way and better for organizational purposes (like adding a language filter to the content item picker). Also my original hope was, that it might just work without any extra effort.

Unfortunately just removing the LocalizationPart from the menu definition does not do the job...
The old localized menu shows the same behavior - creating a new one is only possible as draft. Publishing claims to be successful, but is not. :-(

**Reply by bashuss** [reply]
Ok, I found the error ... for some unknown reason, this
the newly created menu shape renders nothing and a fallback takes over.
So it may have worked in the first place, if `{{ menu | shape_render }}` would create any output, but it does not.
I don't know why yet, as the same kind of code works for me at other locations...
